source/client/Controller.py

```diff
@@ -95,3 +99,33 @@ def Network_discardInfo(self, data):
         top_card = Card.deserialize(data["top_card"])
         size = data["size"]
         self._state.updateDiscardInfo(top_card, size)
+
+    ### Check user's actions, and remind them of rules as necessary ###
+    def discardLogic(self, confirmed, discards):
```

Comment:
So  - this method has unecessary parts and needs rework.
1) empty discard when zeaphod is never submitted - clientState/controller handle ending turn when you play the last card in your hand.
2) We should separate size checking discards before confirmation and actually removing the card form the hand after confirmation into 2 methods, DiscardRuleCheck and DiscardSubmit, in the controller.
Calling them should be done in a HandleDiscard or DiscardLogic helper in the handView which is called from in the eventLoop, so you never leave the eventLoop when doing ruleCheck, confirmation, submission process.
Other option is to just submit, and we raise exception on failed rule checks. (no confirmation or confirmation before rulecheck).

Reply:
We've asked for feedback on how to implement this.  Until that's resolved, I'm only making edits we agreed on.  
Removed Zaephod check -- we'll want to send note to handview when len(current_hand) = 0 so that user will see "Turn is over"  so they know they haven't really gone out.